main.tf

```diff
@@ -142,6 +144,30 @@ resource "aws_s3_bucket" "bucket" {
     }
   }
 
+  dynamic "replication_configuration" {
+    for_each = local.replication
+    iterator = replication
+
+    content {
+      role = join("", aws_iam_role.replication.*.arn)
+
+      rules {
+        id     = "bucket_replication"
+        status = "Enabled"
```

Comment:
should be set by rule i assume as we want to support more than one rule?

Reply:
can you replace lookup with try() too?